core/conversion/converters/impl/batch_norm.cpp

```diff
@@ -40,17 +41,9 @@ auto batch_norm_registrations TRTORCH_UNUSED = RegisterNodeConversionPatterns().
       LOG_DEBUG("training disregarded");
       LOG_DEBUG("cudnn disregarded");
 
-      auto should_unpack = util::toVec(orig_shape).size() < 4;
-      if (should_unpack) {
-        // expand spatial dims from 1D to 2D
-        auto new_shape = util::toDimsTailPad(util::toVec(orig_shape), 4);
-        LOG_DEBUG(
-            "Input shape is less than 4D got: "
-            << orig_shape << ", inserting shuffle layer to reshape to 4D tensor shape: " << new_shape);
-        auto in_shuffle = ctx->net->addShuffle(*input);
-        in_shuffle->setReshapeDimensions(new_shape);
-        in_shuffle->setName(std::string("[Reshape input to " + util::toStr(new_shape) + ']').c_str());
-        input = in_shuffle->getOutput(0);
+      auto expandDims = addPaddingLayer(ctx, n, input, 4);
```

Comment:
I see why you did `ILayer*/nullptr` here, i feel like we should maybe have people explicitly decide to pad based on their own logic and so addPaddingLayer will always give you a layer/tensor back. Either way we need to have logic on if padding should/has happened but in this case, you might get a `nullptr` back which if not handled properly could cause problems.

Reply:
Agreed about people better decide explicitly.